crates/marmot-c/examples/smoke.php

```
@@ -0,0 +1,411 @@
+     *
+     * @param list<string>       $strings
+     * @param list<\FFI\CData>   $buffers out-param: backing buffers to pin
+     */
+    private static function marshalStrings(FFI $ffi, array $strings, array &$buffers): FFI\CData
+    {
+        $count = max(count($strings), 1); // char*[0] is not a valid type
+        $array = $ffi->new("char*[$count]");
+        foreach (array_values($strings) as $i => $s) {
+            $buf = $ffi->new('char[' . (strlen($s) + 1) . ']', false);
+            FFI::memcpy($buf, $s, strlen($s));
+            $buffers[] = $buf; // keep alive across the call
```

PHPStan (2.2.2)

[error] Parameter &$buffers by-ref type of method Marmot::marshalStrings() expects list<FFI\CData>, non-empty-list<FFI\CData|null> given.
You can change the parameter out type with `@param-out` PHPDoc tag.

(parameterByRef.type)